### PR TITLE
simple-hub: depositing

### PR DESCRIPTION
**Pull request by kerzhner**
The main goal of this PR: restrict _simple-hub_ to only match deposits when `amountDeposited` == `destinationHoldings`.

To simplify tests for this new code, the chain watcher is refactored from callbacks to observables. Note, `ganache-cli` is added as a dev dependency to the `simple-hub` package to enable starting unit tests that rely on ganache from VSCode.

### CODE REVIEW

File: packages/simple-hub/package.json [hunk 1]
```diff
@@ -34,6 +34,7 @@
     "eslint-plugin-jest": "23.6.0",
     "eslint-plugin-prettier": "3.1.2",
     "firebase-server": "^1.1.0",
+    "ganache-cli": "^6.9.1",
```
Comment (snario) [inline packages/simple-hub/package.json]: Maybe ensure it's the same version as other packages to reduce unnecessary `node_modules` bloat

File: packages/simple-hub/src/blockchain/__chain-test__/deposit.test.ts [hunk 2]
```diff
@@ -1,26 +1,34 @@
...
-  AssetHolderWatcherEvent
-} from '../asset-holder-watcher';
+import {assetHolderObservable} from '../asset-holder-watcher';
 import {Blockchain} from '../transaction';
+import {first, pairwise} from 'rxjs/operators';
+import {attachToChainObservable} from '../../wallet/chain-event';
 
 jest.setTimeout(20000);
-const five = bigNumberify(5).toHexString();
+const zero = bigNumberify(0);
```
Comment (snario) [inline packages/simple-hub/src/blockchain/__chain-test__/deposit.test.ts]: This is `ethers.constants.Zero`

File: packages/simple-hub/src/blockchain/asset-holder-watcher.ts [hunk 3]
```diff
@@ -1,33 +1,33 @@
...
-  Deposited
-}
-
-export interface AssetHolderWatcherEvent {
+export interface AssetHolderEvent {
   channelId: string;
-  amountDeposited: string;
-  destinationHoldings: string;
+  amountDeposited: BigNumber;
+  destinationHoldings: BigNumber;
```
Comment (snario) [inline packages/simple-hub/src/blockchain/asset-holder-watcher.ts]: Why switch to `BigNumber`?

File: packages/simple-hub/package.json (hunk 1 above)
Comment (kerzhner) [reply packages/simple-hub/package.json]: Good call. 8e9f64018

File: packages/simple-hub/src/blockchain/__chain-test__/deposit.test.ts (hunk 2 above)
Comment (kerzhner) [reply packages/simple-hub/src/blockchain/__chain-test__/deposit.test.ts]: 8e9f64018

File: packages/simple-hub/src/blockchain/asset-holder-watcher.ts (hunk 3 above)
Comment (kerzhner) [reply packages/simple-hub/src/blockchain/asset-holder-watcher.ts]: I find it easier to do arithmetic on `BigNumber`s. Not that we are doing much of that on the _simple-hub_ at the moment. Any objections?

File: packages/simple-hub/src/blockchain/eth-asset-holder.ts [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  private static async attachEthAssetHolder() {
+    if (Blockchain.ethAssetHolder) {
+      return Blockchain.ethAssetHolder;
+    }
+    const newAssetHolder = await createEthAssetHolder();
+    Blockchain.ethAssetHolder = Blockchain.ethAssetHolder || newAssetHolder;
+    return Blockchain.ethAssetHolder;
+  }
+
+  private;
```
Comment (tomclose) [inline packages/simple-hub/src/blockchain/eth-asset-holder.ts]: ?

Comment (kerzhner) [reply packages/simple-hub/src/blockchain/eth-asset-holder.ts]: An editing mistake.

**Pull request merged**